### PR TITLE
Helping out with the documentation :)

### PR DESCRIPTION
**Pull request by corroded**
I've read the guide for writing a scenario , and I'm still a bit of confused as to what is actually needed. I've added a tags feature scenario since it's the one feature I always use. Please have a look and see if this is sufficient enough so I can contribute more scenarios for it :) Thanks!

### CODE REVIEW

**Comment by jbpros** [comment]
Thank you Eumir! This is highly appreciated.

I think you can aim at a higher level of abstraction to be even less imperative. We really want a pure declarative form so that we can run all features against any implementation. For example, cucumber.js running in a browser does not know anything about files or CLI's. This means that steps like `Given a file named "features/tags.feature" with:` are not abstract enough.

You could write something like

```
Given the following feature:
  """
  Feature...
  """
And the following feature:
...
```

The matching step definition will be responsible for storing it (in memory, on disk or whatever).

This is also true for the way you "run" cucumber. Here is an idea:

```
When I run cucumber with tags "@sample-tag"
```

Again, the step definition will take care of the way to actually do that on its specific implementation.

What do you think?

**Comment by corroded** [comment]
I think this is an excellent suggestion! I overlooked cucumber-js and was basing my example from a previous legacy feature so pardon that error. I will be pushing another commit in awhile. Are there any other fixes to the scenario or should I continue in adding up more scenarios?

File: tags.feature [hunk 1]
```diff
@@ -0,0 +1,65 @@
+Feature: Tags feature
```
Comment (jbpros) [inline tags.feature]: Just a little detail here: `Feature: Tags` should be sufficient, right? :)

**Comment by jbpros** [comment]
This is great! 

A little note concerning the expectation step `Then it should pass with exactly:`: we are currently moving from pretty to progress formatter on cucumber.js/global features. This is stil open to discussion though; I am aware that it might not be sufficient for certain test cases (like ensuring the correct scenarios are run based on tags!). We might even have to build a special listener/formatter for running the features.

Don't change anything just yet, I suggest we update it when the progress formatter is ready on the JS implementation and we have a clear decision about the way we check feature results.

**Comment by corroded** [comment]
Okay fixed! So with that note in mind, should I continue with adding more new scenarios until the progress formatter is ready or should we wait for them?

**Comment by jbpros** [comment]
Please add more scenarios if you're willing to! We'll take care of the expectations syntax later on. 

Merging this pull request now. 

Thank you again!

**Comment by jbpros** [comment]
Actually, there is an issue I didn't spot. Sorry about that.

We try to stick to a limited set of cucumber functionalities within global features. In other terms, global features should only use "core" feature elements, those described within [core.feature](https://github.com/cucumber/cucumber-features/blob/master/core.feature). See [cucumber-features rule #5](https://github.com/cucumber/cucumber-features/wiki/Rules)

And the tags.feature you submitted includes a Background, which is not part of core. A solution to this is to copy your context steps to each scenario.

If this is not acceptable to you, I'd be glad to discuss it.

**Comment by corroded** [comment]
Hmm I also missed that point. I have no problems with that, lemme just fix that in a bit. Also I have found an issue which I am not actually sure of. How do you AND tags in cucumber.js or any other implementation? I've added the OR scenario. Also, what about the special tags like no-txn, javascript, or allow-rescue?

**Comment by jbpros** [comment]
Sorry for replying late, had difficulties finding a decent connection here in London.

Regarding the tag issue, I suggest you don't use an option string, bur instead write some declarative steps like this:

```
When I run cucumber including tags "tag1", "tag2"

When I run cucumber excluding tags "tag3", "tag4"

When I run cucumber including tags "tag1", "tag2" and excluding tags "tag3", "tag4"
```

So that we don't have to think about how those are passed to cucumber implementations.

WDYT?

**Comment by corroded** [comment]
That's a great idea! I feel stupid for not thinking about that. I was trying to think of a way the other day but my mind was set at looking at the tags documentation to I guess I was influenced by that. Anyway I'll do a quick fix of that. BTW, there has been some discussion about the format of the features by Andrew. Have you checked the mailing list?

**Comment by jbpros** [comment]
No problem, you're doing a great work man!

I just answered on the ML, I'm kinda off the grid until tomorrow. Not easy to react in those conditions ;)

**Comment by corroded** [comment]
Changed the wording here. Like discussed in the groups do we need to add more detailed features here? It's a bit hard to understand if you're just solely looking in the features file for cuke reference because if it is too abstract, the user will not know the proper syntax of say, excluding a tag(unless he looks in the step definition). Was this the intended purpose?

**Comment by jbpros** [comment]
Thank you for the fix.

Regarding the current discussions about feature syntax, I'd say: let's keep it like this for now. We are still experimenting, trying to find the right level of abstraction and the best compromise in terms of "declarativeness versus documenting efficiently".

I invite you to follow and contribute to the discussion on the group if you're interested.

**Comment by corroded** [comment]
Yup I'll try to find my words when I can :) Anyway, should I continue with other tag features or will this pull request do for now?

**Comment by jbpros** [comment]
Would you mind holding the work here for a few days? We've been discussing other approaches and it is very likely that we will have to rewrite things again. That's why I won't merge into master yet. But it will, eventually!

Of course, if you don't mind rewriting stuff several times, you might want to go on and capture more use cases :)

**Comment by corroded** [comment]
No problem! I'll see if I can add more use cases, also for the other features

**Comment by jbpros** [comment]
This has been superseded by a more declarative [_tags_ feature file](https://github.com/cucumber/cucumber-tck/commit/308240ef37151d0a3cd50f02a545c0e1e8764043).

@corroded Feel free to comment on that one.

**Pull request closed without merging**